src/jit/flowgraph.cpp

```diff
@@ -5746,14 +5746,36 @@ void          Compiler::fgFindBasicBlocks()
 
     if (compIsForInlining())
     {
+        bool hasReturnBlocks = false;
+        bool hasMoreThanOneReturnBlock = false;
+
+        for (BasicBlock* block = fgFirstBB; block; block = block->bbNext)
+        {
+            if (block->bbJumpKind == BBJ_RETURN)
+            {
+                if (hasReturnBlocks)
+                {
+                    hasMoreThanOneReturnBlock = true;
+                    break;
+                }
+
+                hasReturnBlocks = true;
+            }
+        }
+
+        if (!hasReturnBlocks)
+        {
+            impInlineInfo->inlineResult->NoteFatal(InlineObservation::CALLEE_DOES_NOT_RETURN);
+            return;
+        }
+
```

Comment:
I worry that this might be too general. I've seen cases where people have written performance-sensitive code that is effectively `while (true) ...` and then relied on exceptions to terminate processing, and such methods can benefit from being inlined.

Ideally we'd like to check that the inlinee does very little other than throw an exception.